### PR TITLE
Unfold / decode correctly upon ES indexing

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[Building Java 8 aj-31-0fe7a4c4-7255-4b22-8331-3b3cb00bbbd5] Failed tests: 
[Building Java 8 aj-31-0fe7a4c4-7255-4b22-8331-3b3cb00bbbd5]   CassandraGetMessageListMethodTest>GetMessageListMethodTest.getMessageListShouldNotFailWhenHeaderIsValid:536 1 expectation failed.
[Building Java 8 aj-31-0fe7a4c4-7255-4b22-8331-3b3cb00bbbd5] Expected status code <200> doesn't match actual status code <401>.
```

test this please

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java [hunk 1]
```diff
@@ -94,7 +104,27 @@ public void addressWithTwoDisplayNamesOnTheSameFieldShouldBeRetrieved() {
         assertThat(headerCollection.getFromAddressSet())
             .containsOnly(new EMailer("Christophe Hamerling", "chri.hamerling@linagora.com"),
                 new EMailer("Graham CROSMARIE", "grah.crosmarie@linagora.com"));
+    }
+
+    @Test
+    public void foldedFromHeaderShouldBeSupported() {
+        HeaderCollection headerCollection = HeaderCollection.builder()
+            .add(new FieldImpl("From", "Christophe Hamerling <chri.hamerling@linagora.com>,\r\n Graham CROSMARIE <grah.crosmarie@linagora.com>"))
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: can you put this string on two lines to clarify the intent please?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java [hunk 2]
```diff
@@ -94,7 +104,27 @@ public void addressWithTwoDisplayNamesOnTheSameFieldShouldBeRetrieved() {
...
+
+        assertThat(headerCollection.getFromAddressSet())
+            .containsOnly(new EMailer("Christophe Hamerling", "chri.hamerling@linagora.com"),
+                new EMailer("Graham CROSMARIE", "grah.crosmarie@linagora.com"));
+    }
+
+    @Test
+    public void foldedHeaderShouldBeSupported() {
+        HeaderCollection headerCollection = HeaderCollection.builder()
+            .add(new FieldImpl("From", "Christophe Hamerling <chri.hamerling@linagora.com>,\r\n Graham CROSMARIE <grah.crosmarie@linagora.com>"))
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: idem

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java [hunk 3]
```diff
@@ -32,52 +32,52 @@
     @Test
     public void simpleValueAddressHeaderShouldBeAddedToTheAddressSet() {
         HeaderCollection headerCollection = HeaderCollection.builder()
-            .add(new FieldImpl("To", "btellier@linagora.com"))
```
Comment (strokyl) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: Those address will remain in git history anyway..

**Comment by strokyl** [comment]
Ok we put header in ES but we do not specify anything about them in the mapping !!!
By default each time ES see a field (or "subfield") it infear the type for it and create a index for it. If a new document arrive with the same field with a type that is different from the inferre type that will create a exception.
So please either specify all the header we can encounter in the mapping and explain me why we need to index them or either change the mapping to dynamic = false so ES ignore field not specified in the mapping.

**Comment by chibenwa** [comment]
@strokyl  that is a nice PR comment.

However, it is not related to the PR here, which is just about decoding/unfolding headers. I agree that further work might be needed for this, but I don't see why I would have to do it here.

**Pull request closed without merging**